### PR TITLE
Update Vale guidelines checker and fix issues

### LINKED ISSUE #560: Update Vale
Vale CI guidelines checker is not the latest version

See https://github.com/errata-ai/vale/releases

### PR DESCRIPTION
**Pull request by NicolasMassart**
**Before pushing any commit in your pull request, make sure that:**

- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.
- [x] you verified the rendering on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs) (see [Testing](#testing))

## Describe the change

- Update Vale to latest version https://github.com/errata-ai/vale/releases/tag/v2.6.6 (required to add all styles in the project as they are not included in Vale natively anymore)
- fixes the issues checked by styles
- add words in accepted list
- exclude Vale styles from markdown linting (not our docs)

## Issue fixed

fixes #560 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [x] CircleCI workflow
- [x] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] Github integration

## Review process

You can probably skip looking at the changes in `CI/vale/vale_styles/Microsoft`, `CI/vale/vale_styles/proselint` and `CI/vale/vale_styles/write-good` as they are imported styles I did not change them only moved them or added some.
Only the `CI/vale/vale_styles/Besu` and `CI/vale/vale_styles/Vocab/Besu/accept.txt` have changes that you should check.

## Testing

Run ` CI/scripts/test_all.sh `

### CODE REVIEW

File: CI/vale/vale_styles/Microsoft/ComplexWords.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+extends: substitution
```
Comment (bgravenorst) [inline CI/vale/vale_styles/Microsoft/ComplexWords.yml:1]: Can this rule also perhaps flag "prior to", and suggest using "before"?

File: CI/vale/vale_styles/proselint/Diacritical.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  Ubermensch: Übermensch
+
+  # Swedish loanwords
+  filmjolk: filmjölk
+  smorgasbord: smörgåsbord
+
+  # Names, places, and companies
+  Beyonce: Beyoncé
+  Bronte: Brontë
+  Bronte: Brontë
```
Comment (bgravenorst) [inline CI/vale/vale_styles/proselint/Diacritical.yml:83]: duplication?

File: docs/Concepts/Pruning.md [hunk 3]
```diff
@@ -12,8 +12,8 @@ latest state for the blockchain (and some older states, but not all). You can ch
 balances, sign and send transactions, and look at current Dapp data.
 
 Archive nodes have all of this and they also store the intermediary state of every account and
-contract for every block since the genesis block. With an archive node you can do everything you
-do with a full node, as well as access historical state data.
+contract for every block since the genesis block.
+With an archive node you can do everything you do with a full node and also access historical state data.
```
Comment (bgravenorst) [inline docs/Concepts/Pruning.md:16]: Suggestion, "An archive can do everything a full node does, and it can access historical state data."

**Review by bgravenorst — APPROVED** [review]
Minor suggestion.

File: docs/Concepts/Privacy/Multi-Tenancy.md [hunk 4]
```diff
@@ -30,8 +30,8 @@ a _tenant_, and the operator is the _owner_ of the Besu and Orion node.
     trust mode.
 
 Multi-tenancy validates that tenants have permission to use the specified HTTP or Websocket
-JSON-RPC requests, and the tenant has access to the requested privacy data. There is segregation of
-private data, and each tenant uses a JSON Web Token (JWT) for authentication.
+JSON-RPC requests, and the tenant has access to the requested privacy data.
+Private data are isolated and each tenant uses a JSON Web Token (JWT) for authentication.
```
Comment (MadelineMurray) [inline docs/Concepts/Privacy/Multi-Tenancy.md:34]: are isolated -> is isolated

File: docs/Concepts/Pruning.md (hunk 3 above)
Comment (MadelineMurray) [inline docs/Concepts/Pruning.md:16]: node and also -> node, and also

File: docs/HowTo/Upgrade/Upgrade-Protocol.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ description: Upgrading protocol versions
 
 # Upgrading your protocol in a private network
 
-To [upgrade the protocol](../../Concepts/Protocol-Upgrades.md) (also known as a hardfork) in a
+To [upgrade the protocol](../../Concepts/Protocol-Upgrades.md) (also known as a "hard fork") in a
```
Comment (MadelineMurray) [inline docs/HowTo/Upgrade/Upgrade-Protocol.md:7]: Suggest removing the "" around hard fork - I don't think they're needed

**Review by MadelineMurray — APPROVED** [review]
Couple of minor comments.  Nice to see all those spelling errors fixed :-)

File: CI/vale/vale_styles/proselint/Diacritical.yml (hunk 2 above)
Comment (NicolasMassart) [reply CI/vale/vale_styles/proselint/Diacritical.yml:83]: It's an imported file, not written by us, I did not check that... Si I will ignore ;)

File: docs/Concepts/Pruning.md (hunk 3 above)
Comment (NicolasMassart) [reply docs/Concepts/Pruning.md:16]: solved using @bgravenorst suggestion

**Pull request merged**